File: web/src/api/group.ts
import request from '@/utils/request'
import type { ApiResponse, PagedResponse } from '@/types/models'

// 分组模型
export interface Group {
  id: number
  tenant_id?: string
  name: string
  alias?: string
  description?: string
  member_count?: number
  created_at: string
  updated_at: string
}

// 分组成员
export interface GroupMember {
  id: number
  group_id: number
  user_id: number
  created_at: string
  user?: {
    id: number
    name: string
    alias?: string
    role: string
  }
}

const tenantHeaders = (tenantId?: string) => tenantId
  ? { headers: { 'X-Tenant-ID': tenantId } }
  : {}

// 获取分组列表
export const getGroups = (params?: { tenant_id?: string; search?: string; page?: number; size?: number }) => {
  return request.get<any, PagedResponse<Group[]>>('/api/v1/admin/groups', { params, ...tenantHeaders(params?.tenant_id) })
}

// 获取分组详情
export const getGroup = (id: number, tenantId?: string) => {
  return request.get<any, ApiResponse<Group>>(`/api/v1/admin/groups/${id}`, tenantHeaders(tenantId))
}

// 创建分组
export const createGroup = (data: { tenant_id?: string; name: string; alias?: string; description?: string }) => {
  return request.post<any, ApiResponse<Group>>('/api/v1/admin/groups', data, tenantHeaders(data.tenant_id))
}

// 更新分组
export const updateGroup = (id: number, data: { name?: string; alias?: string; description?: string }, tenantId?: string) => {
  return request.put<any, ApiResponse<Group>>(`/api/v1/admin/groups/${id}`, data, tenantHeaders(tenantId))
}

// 删除分组
export const deleteGroup = (id: number, tenantId?: string) => {
  return request.delete<any, ApiResponse>(`/api/v1/admin/groups/${id}`, tenantHeaders(tenantId))
}

// 获取分组成员
export const getGroupMembers = (groupId: number, tenantId?: string) => {
  return request.get<any, ApiResponse<GroupMember[]>>(`/api/v1/admin/groups/${groupId}/members`, tenantHeaders(tenantId))
}

// 添加分组成员
export const addGroupMember = (groupId: number, userId: number, tenantId?: string) => {
  return request.post<any, ApiResponse>(`/api/v1/admin/groups/${groupId}/members`, { user_ids: [userId] }, tenantHeaders(tenantId))
}

// 移除分组成员
export const removeGroupMember = (groupId: number, userId: number, tenantId?: string) => {
  return request.delete<any, ApiResponse>(`/api/v1/admin/groups/${groupId}/members/${userId}`, tenantHeaders(tenantId))
}

File: internal/server/model/group.go
package model

import "time"

// Group 统一分组模型
// 合并 AgentGroup 和 ClientGroup，统一管理所有用户分组
type Group struct {
	ID          int64     `gorm:"primaryKey" json:"id"`
	TenantID    string    `gorm:"size:36;index;uniqueIndex:uk_group_tenant_name,priority:1" json:"tenant_id,omitempty"`
	Name        string    `gorm:"size:100;not null;uniqueIndex:uk_group_tenant_name,priority:2" json:"name"`
	Alias       string    `gorm:"size:100" json:"alias"`       // 别名（显示名称）
	Description string    `gorm:"size:500" json:"description"` // 描述
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	// 关联
	Members []*GroupMember `gorm:"foreignKey:GroupID" json:"members,omitempty"`
}

func (Group) TableName() string {
	return "group"
}
